pkg/sdk/sdk_test.go

```diff
@@ -0,0 +1,518 @@
+			setupFunc: func() *httptest.Server {
+				svr := testServer("", http.MethodPut, func(w http.ResponseWriter, r *http.Request) {})
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if err.Error() != "non success status code received : 404" {
+					t.Errorf("Want: %v, Got: %v", "non success status code received : 404", err.Error())
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure:: Replace :: http post failure",
+			setupFunc: func() *httptest.Server {
+				svr := testServer("/v1/register{id}", http.MethodPut, func(w http.ResponseWriter, r *http.Request) {})
+				svr.Close()
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if !strings.Contains(err.Error(), "No connection could be made because the target machine actively refused it") {
+					t.Errorf("Want: %v, Got: %v", "No connection could be made because the target machine actively refused it.", err.Error())
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			svr := tt.setupFunc()
+			defer tt.cleanupFunc(svr)
+
+			calls := NewHtmlToPdfSvc(svr.URL)
+
+			err := calls.Replace([]byte("abc"), tt.id)
+
+			tt.ValidateFunc(err)
+		})
+	}
+}
+func Test_GeneratePdf(t *testing.T) {
+	type Student struct {
+		Name  string
+		Marks int
+		Id    string
+	}
+	type Class []Student
+	var class Class
+	// defining struct instance
+	std1 := Student{"A", 90, "1"}
+	std2 := Student{"B", 100, "2"}
+	std3 := Student{"C", 88, "3"}
+	std4 := Student{"D", 25, "4"}
+	std5 := Student{"E", 35, "5"}
+	class = append(class, std4, std2, std3, std1, std5)
+	tests := []struct {
+		name              string
+		id                string
+		data              map[string]interface{}
+		setupFunc         func() *httptest.Server
+		mockServerHandler func(w http.ResponseWriter, r *http.Request)
+		ValidateFunc      func(b []byte, err error)
+		cleanupFunc       func(*httptest.Server)
+		expectedResponse  model.Response
+	}{
+		{
+			name: "Success:: GeneratePdf",
+			id:   "1",
+			data: map[string]interface{}{"id": "1"},
+			setupFunc: func() *httptest.Server {
+				svr := testServer("/v1/generate/{id}", http.MethodPost, func(w http.ResponseWriter, r *http.Request) {
+					vars := mux.Vars(r)
+					//we take id as a parameter from url path
+					id, ok := vars["id"]
+					if !ok {
+						response.ToJson(w, http.StatusBadRequest, "id not found", nil)
+						return
+					}
+					var data GenerateReq
+					err := json.NewDecoder(r.Body).Decode(&data)
+					if err != nil {
+						response.ToJson(w, http.StatusBadRequest, "error decoding data", nil)
+						t.Error(err.Error())
+						return
+					}
+					data.Id = id
+					if err != nil {
+						response.ToJson(w, http.StatusBadRequest, "file read fail", nil)
+						t.Error(err.Error())
+						return
+					}
+
+					if !reflect.DeepEqual(data.Values["id"], "1") {
```

Comment:
It looks like the entire struct is being populated.
Why not compare the entire struct?

Reply:
done